### PR TITLE
fix: AU-1094: Address component requirements

### PR DESCRIPTION
**Pull request by jiisuominen**
# [AU-1094](https://helsinkisolutionoffice.atlassian.net/browse/AU-1094)

Address component did not work as expected with field requirements. This fixes it.

Also some "attachmentIsNew" fixes, which just became obsolete. Will leave them for posterity.

## What was done

* Fix address requirements.

[AU-1094]: https://helsinkisolutionoffice.atlassian.net/browse/AU-1094?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: public/modules/custom/grants_metadata/src/TypedData/Definition/GrantsAttachmentDefinition.php [hunk 1]
```diff
@@ -48,7 +48,7 @@ public function getPropertyDefinitions(): array {
           'fileType',
         ]);
 
-      $info['integrationID'] = DataDefinition::create('string')
+      $info['integrationID'] = DataDefinition::create('boolean')
```
Comment (oskarimysteerio) [inline public/modules/custom/grants_metadata/src/TypedData/Definition/GrantsAttachmentDefinition.php:51]: Integration ID tuskin boolean?

File: public/modules/custom/grants_attachments/src/Element/GrantsAttachments.php [hunk 2]
```diff
@@ -193,6 +198,10 @@ public static function processWebformComposite(&$element, FormStateInterface $fo
         $element["fileStatus"]["#value"] = 'justUploaded';
       }
     }
+    else{
+
+      $d = 'asdf';
```
Comment (oskarimysteerio) [inline public/modules/custom/grants_attachments/src/Element/GrantsAttachments.php:203]: le debug

**Review by oskarimysteerio — CHANGES_REQUESTED** [review]

File: public/modules/custom/grants_metadata/src/TypedData/Definition/GrantsAttachmentDefinition.php (hunk 1 above)
Comment (jiisuominen) [reply public/modules/custom/grants_metadata/src/TypedData/Definition/GrantsAttachmentDefinition.php:51]: hups.

joo ne oli väärin tuolla ja näköjään innostunu liikaa kopsimaan. mitenköhän menny ohi tossa.

File: public/modules/custom/grants_attachments/src/Element/GrantsAttachments.php (hunk 2 above)
Comment (jiisuominen) [reply public/modules/custom/grants_attachments/src/Element/GrantsAttachments.php:203]: hups v2

**Review by strangeworlder — APPROVED** [review]

**Pull request merged**